src/app/api/loan/route.ts

```diff
@@ -0,0 +1,23 @@
+import { NextResponse, NextRequest } from "next/server";
+import connectDatabase from "@/lib/database";
+import LoanModel from "@/models/loan-model";
+
+export async function POST(request: NextRequest) {
+
+    const body = await request.json();
+    if (!body.userAddress) {
+        return NextResponse.json({ message: "User wallet address is required" }, { status: 400 });
+    }
```

Comment:
This validation is far not enough for an API

Lets use express validator, you need to validate each parameter of the JSON (which I think they are several)

https://yaali4422519.medium.com/next-js-validate-request-using-next-connect-and-express-validator-464997067a9c

Reply:
it was just a placeholder  though, not fully implemented